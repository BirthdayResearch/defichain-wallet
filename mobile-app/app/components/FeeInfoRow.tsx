import { tailwind } from '@tailwind'
import { ThemedText, ThemedView } from './themed'
import { translate } from '@translations'
import { View } from 'react-native'
import NumberFormat from 'react-number-format'
import { BottomSheetInfo } from '@components/BottomSheetInfo'

interface FeeInfoRowProps {
  value: string | number
  type: FeeType
  suffix?: string
  testID: string
}

type FeeType = 'ESTIMATED_FEE' | 'VAULT_FEE'

export function FeeInfoRow (props: FeeInfoRowProps): JSX.Element {
  const estimatedFee = {
    title: 'Estimated fee',
    message: 'Each transaction will be subject to a small amount of fees. The amount may vary depending on the networkâ€™s congestion.'
  }
  const vaultFee = {
    title: 'Vault fee',
    message: 'This fee serves as initial deposit for your vault. You will receive 1 DFI back when you choose to close this vault.'
  }

  return (
    <ThemedView
      dark={tailwind('bg-gray-800 border-b border-gray-700')}
      light={tailwind('bg-white border-b border-gray-200')}
      style={tailwind('p-4 flex-row items-start w-full')}
    >
      <View style={tailwind('w-5/12')}>
        <View style={tailwind('flex-row items-center justify-start')}>
          <ThemedText style={tailwind('text-sm mr-1')} testID={`${props.testID}_label`}>
            {translate('components/BottomSheetInfo', props.type === 'ESTIMATED_FEE' ? estimatedFee.title : vaultFee.title)}
          </ThemedText>
          <BottomSheetInfo alertInfo={props.type === 'ESTIMATED_FEE' ? estimatedFee : vaultFee} name={props.type === 'ESTIMATED_FEE' ? estimatedFee.title : vaultFee.title} />
        </View>
      </View>

      <View style={tailwind('flex-1 flex-row justify-end flex-wrap items-center')}>
        <NumberFormat
          decimalScale={8}
          displayType='text'
          renderText={(val: string) => (
            <ThemedText
              dark={tailwind('text-gray-400')}
              light={tailwind('text-gray-500')}
              style={tailwind('text-sm text-right')}
              testID={props.testID}
            >
              {val}
            </ThemedText>
          )}
          thousandSeparator
          value={props.value}
        />
        <ThemedText
          light={tailwind('text-gray-500')}
          dark={tailwind('text-gray-400')}
          style={tailwind('text-sm ml-1')}
          testID={`${props.testID}_suffix`}
        >
          {props.suffix}
        </ThemedText>
      </View>
    </ThemedView>
  )
}
