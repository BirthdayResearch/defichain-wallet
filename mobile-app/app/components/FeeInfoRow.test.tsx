import { render } from '@testing-library/react-native'
import { FeeInfoRow } from './FeeInfoRow'

jest.mock('@shared-contexts/ThemeProvider')

jest.mock('./BottomSheetModal', () => ({
  BottomSheetModal: () => <></>
}))

describe('estimated fee info scanner', () => {
  it('should match snapshot of estimated fee', async () => {
    const component = (
      <FeeInfoRow
        type='ESTIMATED_FEE'
        value='100'
        testID='text_fee'
        suffix='DFI'
      />
    )
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('should match snapshot of vault fee', async () => {
    const component = (
      <FeeInfoRow
        type='VAULT_FEE'
        value='100'
        testID='text_fee'
        suffix='DFI'
      />
    )
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
