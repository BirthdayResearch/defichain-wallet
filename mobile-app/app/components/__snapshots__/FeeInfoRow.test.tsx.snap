// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`estimated fee info scanner should match snapshot of estimated fee 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "41.666667%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 14,
                "lineHeight": 20,
                "marginRight": 4,
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="text_fee_label"
      >
        Estimated fee
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "flexWrap": "wrap",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 20,
              "textAlign": "right",
            },
            Object {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="text_fee"
    >
      100
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 20,
              "marginLeft": 4,
            },
            Object {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="text_fee_suffix"
    >
      DFI
    </Text>
  </View>
</View>
`;

exports[`estimated fee info scanner should match snapshot of vault fee 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "41.666667%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 14,
                "lineHeight": 20,
                "marginRight": 4,
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="text_fee_label"
      >
        Vault fee
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "flexWrap": "wrap",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 20,
              "textAlign": "right",
            },
            Object {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="text_fee"
    >
      100
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 20,
              "marginLeft": 4,
            },
            Object {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="text_fee_suffix"
    >
      DFI
    </Text>
  </View>
</View>
`;
